seam/seam.py

```diff
@@ -25,19 +40,45 @@ def __init__(
         endpoint: Optional[str] = None,
         wait_for_action_attempt: Optional[Union[bool, Dict[str, float]]] = True,
     ):
-        """
-        Parameters
-        ----------
-        api_key : str, optional
-          API key.
-        personal_access_token : str, optional
-          Personal access token.
-        workspace_id : str, optional
-          Workspace id.
-        endpoint : str, optional
-          The API endpoint to which the request should be sent.
-        wait_for_action_attempt : bool or dict, optional
-          Controls whether to wait for an action attempt to complete, either as a boolean or as a dictionary specifying `timeout` and `poll_interval`. Defaults to `False`.
+        """Initialize a Seam client instance.
+
+        This method sets up the Seam client with the provided authentication credentials and configuration options.
+        It supports two authentication methods: API key or personal access token.
+
+        :param api_key: The API key for authenticating with Seam. Mutually
+            exclusive with personal_access_token
+        :type api_key: Optional[str]
+        :param personal_access_token: A personal access token for
+            authenticating with Seam. Mutually exclusive with api_key
+        :type personal_access_token: Optional[str]
+        :param workspace_id: The ID of the workspace to interact with.
+            Required when using a personal access token
+        :type workspace_id: Optional[str]
+        :param endpoint: The custom API endpoint URL. If not provided, the
+            default Seam API endpoint will be used
+        :type endpoint: Optional[str]
+        :param wait_for_action_attempt: Controls whether to wait for an
+            action attempt to complete. Can be a boolean or a dictionary with
+            'timeout' and 'poll_interval' keys
+        :type wait_for_action_attempt: Optional[Union[bool, Dict[str, float]]]
+
+        :raises SeamInvalidOptionsError: If neither api_key nor
+            personal_access_token is provided, or if workspace_id is missing
+            when using a personal access token
+        :raises SeamInvalidTokenError: If the provided API key or personal
+            access token format is invalid
+        :raises SeamHttpApiError: For general API errors, including
+            unexpected server responses
+        :raises SeamHttpUnauthorizedError: When the provided authentication
+            credentials (api_key or personal_access_token) are invalid
+        :raises SeamHttpInvalidInputError: When the API request contains
+            invalid input data
+        :raises SeamActionAttemptFailedError: When an action attempt fails to
+            complete successfully (only when wait_for_action_attempt is
+            enabled)
+        :raises SeamActionAttemptTimeoutError: When an action attempt exceeds
+            the specified timeout duration (only when wait_for_action_attempt
+            is enabled)
```

Comment:
Oh yeah I meant to ask about this but forgot. Probably it doesn't make sense to put them all here. Http errors can go to `SeamHttpClient` docstring and action attempt ones to the `ActionAttempts` route class docstring. I think `SeamInvalidOptionsError` and `SeamInvalidTokenError` is fine to leave here.

> I wonder what other libs do, since often errors are raised in many places.

I briefly looked through a couple of libs and it seems that mostly exceptions are documented where they're raised. I also found instances where exception docstring is also propagated up where the method throwing it is used like [here](https://github.com/sqlalchemy/sqlalchemy/blob/96f1172812f858fead45cdc7874abac76f45b339/lib/sqlalchemy/orm/base.py#L384-L424).

Even though `InvalidOptionsError` is raised in `get_auth_headers` which in turn is called in `parse_options` that we use in the Seam constructor, I think it still makes sense to document it on the constructor because none of those functions handle the error and it'll get raised when the constructor is called. Plus if the error is not documented on the constructor, it won't be clear from the IDE intellisense that when calling Seam constructor such errors can be raised. Ideally we should document the error on each level where it can be raised?